### ISSUE TITLE
Override linters settings by conditions

### ISSUE THREAD

**Issue by rugleb** [golangci/golangci-lint#3952]
### Your feature request related to a problem? Please describe.

Now I can configure the linter parameters in the `linters-settings` block and disable it in the `issues` block for tests:

```yml
linters-settings:
  funlen:
    lines: 30

issues:
  exclude-rules:
    - path: _test\.go
      linters:
        - funlen
```

But I can't **override** some linter settings for tests.

### Describe the solution you'd like.

It would be great to add a section to the config where we can specify rules for overriding linter settings:

```yml
linters-settings:
  funlen:
    lines: 30  # 30 lines for runtime

linters-settings-override:
  rules:
    - path: _test\.go
      linters:
          funlen:
              lines: 60  # 60 lines for tests
```

### Describe alternatives you've considered.

No more ideas :)

**Comment by ldez** [comment]
Hello,

the internal design of golangci-lint was not made for this: the settings are used when the linters are created.
Also, it's not possible to override settings without an important extra performance cost.

I consider this like a minor use case, and the design and the performance costs lead me to decline this issue.

I recommend using `exclude` (I know that is not an override).
https://golangci-lint.run/usage/false-positives/#exclude-issues-by-path
